specs/core/0_beacon-chain.md

```diff
@@ -1476,18 +1470,21 @@ def get_genesis_beacon_state(genesis_validator_deposits: List[Deposit],
 
         # Recent state
         latest_crosslinks=[Crosslink(epoch=GENESIS_EPOCH, shard_block_root=ZERO_HASH) for _ in range(SHARD_COUNT)],
-        latest_block_roots=[ZERO_HASH for _ in range(LATEST_BLOCK_ROOTS_LENGTH)],
+        latest_block_roots=[ZERO_HASH for _ in range(SLOTS_PER_BATCHING)],
+        latest_state_roots=[ZERO_HASH for _ in range(SLOTS_PER_BATCHING)],
         latest_active_index_roots=[ZERO_HASH for _ in range(LATEST_ACTIVE_INDEX_ROOTS_LENGTH)],
         latest_slashed_balances=[0 for _ in range(LATEST_SLASHED_EXIT_LENGTH)],
         latest_attestations=[],
-        batched_block_roots=[],
+        historical_batchings=[],
 
         # Ethereum 1.0 chain data
-        latest_eth1_data=latest_eth1_data,
+        latest_eth1_data=genesis_eth1_data,
         eth1_data_votes=[],
         deposit_index=len(genesis_validator_deposits)
     )
 
+    state.latest_block_roots[GENESIS_SLOT % SLOTS_PER_BATCHING] = hash_tree_root(genesis_block_body)
```

Comment:
I would consider filling this array with `hash_tree_root(genesis_block_body)` (similar to the active_index_roots below). The implication here is that the only thing before the genesis_block is the genesis_block. This is more failsafe for any initial state operations that might subtract to a previous epoch/slot.

Reply:
We should initialize `latest_state_roots` in the same way (currently not initialized at all)

Reply:
> This is more failsafe for any initial state operations that might subtract to a previous epoch/slot.

Interesting! I guess we can review this when we finalise the slot/epoch underflow issue.

> the only thing before the genesis_block is the genesis_block

Semantically, the only thing before the `genesis_block` is nothingness, i.e. `ZERO_HASH`.

> currently not initialized at all

`latest_state_root` is initialised to `[ZERO_HASH for _ in range(SLOTS_PER_BATCHING)]`. I think that's the "semantically correct" way of doing it.